### DISCUSSION TITLE
Confusing requests about enforced Spring Security managed authentication module and API routes

### DISCUSSION THREAD

**Discussion by busybox11** [jhipster/generator-jhipster#23809] [Q&A]
I'm currently trying to interface a custom-made module that utilizes Spring Security for authentication, as part of a custom OAuth2 solution.

It is internally exposing a `/login` route on the server via a `RestController`, using `@GetMapping` inside the module. If this can help you, it extends `AbstractAuthenticationProcessingFilter` somewhere with the parameter `"/login"`.

My problem is that I can't seem to be able to access this method from the 8080 port that exposes both the API and the Angular front-end. I am forced to use a `/login` GET callback for OAuth by the module, but everytime I try to use it I am redirected to the frontend, NOT the backend.

Getting the base URL via the provided `applicationConfigService.getEndpointFor` method still redirects to the same base /login URL, and it is still served by Angular.

This is probably very simple - I just got introduced to the project using JHipster. Though I did read all the documentation multiple times, this could indicate a need for more clarifications in the documentation. If it is the case, I apologize for the confusion.

---
EDIT: It looks like it could be related to this issue: https://github.com/jhipster/generator-jhipster/issues/22843

**Comment by TatyOko28** [comment]
What you are seeing is expected behavior in a JHipster gateway/monolith using Angular on port `8080`.

The same server serves:

```text
- Spring Boot backend
- Angular frontend static files
